wgpu-core/src/swap_chain.rs

```diff
@@ -248,40 +248,44 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             .take()
             .ok_or(SwapChainError::AlreadyAcquired)?;
         let (view, _) = hub.texture_views.unregister(view_id.value.0, &mut token);
-        let image = match view.inner {
-            resource::TextureViewInner::Native { .. } => unreachable!(),
-            resource::TextureViewInner::SwapChain { image, .. } => image,
-        };
-
-        let sem = if sc.active_submission_index > device.last_completed_submission_index() {
-            Some(&sc.semaphore)
-        } else {
-            None
-        };
-        let queue = &mut device.queue_group.queues[0];
-        let result = unsafe { queue.present(B::get_surface_mut(surface), image, sem) };
-
-        tracing::debug!(trace = true, "Presented. End of Frame");
-
-        for fbo in sc.acquired_framebuffers.drain(..) {
-            unsafe {
-                device.raw.destroy_framebuffer(fbo);
+        if let Some(view) = view {
```

Comment:
can we just early out here if it's None? I.e. something like
let (view_maybe, _) = hub.texture_views.unregister(view_id.value.0, &mut token);
let view = view_maybe.ok_or(SwapChainError::Invalid)?;
// continue as usual